### PR TITLE
Brodey | adding f.compact

### PR DESCRIPTION
**Pull request by brodeynewman**

### CODE REVIEW

File: packages/compact/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "@foldr/compact",
+  "version": "0.0.0",
+  "license": "ISC",
+  "main": "dist/index.js",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/JasonPollman/foldr.git"
```
Comment (JasonPollman) [inline packages/compact/package.json]: I've been pulling the repo out of these until we have a permanent home.

Comment (JasonPollman) [reply packages/compact/package.json]: Like for an org.

File: packages/compact/src/index.test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const compact = require('./');
```
Comment (JasonPollman) [inline packages/compact/src/index.test.js]: Use import

File: packages/compact/src/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * @param {Array} array - the array to flatten
+ * @returns {Array} a newly flattened array
+ * @memberof foldr
+ * @since v0.0.0
+ * @export
+ * @example
+ * compact([1, null, 3, undefined]);       // => [1, 3]
+ * compact(['', 0, 4]);     // => [4]
+ */
+function compact(array) {
```
Comment (JasonPollman) [inline packages/compact/src/index.js]: For ESM compatibility: `export default`

File: packages/compact/package.json (hunk 1 above)
Comment (brodeynewman) [reply packages/compact/package.json]: ok

File: packages/compact/src/index.test.js (hunk 2 above)
Comment (brodeynewman) [reply packages/compact/src/index.test.js]: ok

File: packages/compact/src/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Flattens an array one level deep
+ * @param {Array} array - the array to flatten
+ * @returns {Array} a newly flattened array
+ * @memberof foldr
+ * @since v0.0.0
+ * @export
+ * @example
+ * compact([1, null, 3, undefined]);       // => [1, 3]
+ * compact(['', 0, 4]);     // => [4]
```
Comment (JasonPollman) [inline packages/compact/src/index.js]: Fix comment spacing.

File: packages/compact/src/index.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  eslint-disable no-array-constructor,
+  one-var,
+  prefer-const,
+  no-unused-vars,
+  one-var-declaration-per-line,
+  no-continue,
+*/
+
+/**
+ * Flattens an array one level deep
```
Comment (JasonPollman) [inline packages/compact/src/index.js]: Fix description.

File: packages/compact/src/index.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ * @file
+ */
+
+/*
+  eslint-disable no-array-constructor,
+  one-var,
+  prefer-const,
+  no-unused-vars,
+  one-var-declaration-per-line,
+  no-continue,
```
Comment (JasonPollman) [inline packages/compact/src/index.js]: Most of these disablers are acutally not used.

**Pull request merged**